src/ucp/wireup/wireup.c

```diff
@@ -559,7 +787,7 @@ static void ucp_wireup_process_request(ucp_worker_h worker, ucp_ep_h ep,
          return;
     }
 
-    if (ep->rsc_index != msg->dst_rsc_index) {
+    if ((msg->dst_rsc_index != (ucp_rsc_index_t)-1) && (msg->dst_rsc_index != ep->rsc_index)) {
```

Comment:
it's a special value in wireup message which means "let destination select its transport instead of the source"